rswag-specs/lib/rswag/specs/request_factory.rb

```diff
@@ -118,6 +118,26 @@ def add_path(request, metadata, swagger_doc, parameters, example)
 
       def build_query_string_part(param, value)
         name = param[:name]
+
+        # OAS 3: https://swagger.io/docs/specification/serialization/
+        if param.has_key?(:schema)
```

Comment:
We should check the doc_version here instead:

```suggestion
        if doc_version(swagger_doc).start_with?('3')
```